holidays/holiday_groups.py

```diff
@@ -0,0 +1,908 @@
+        https://en.wikipedia.org/wiki/Eid_al-Fitr
+        """
+        return self._add_islamic_calendar_holiday(holiday_name, 10, 1)
+
+    def _add_eid_al_fitr_day_two(self, holiday_name) -> List[date]:
+        """
+        Add Eid al-Fitr Day Two.
+
+        https://en.wikipedia.org/wiki/Eid_al-Fitr
+        """
+        return self._add_islamic_calendar_holiday(
+            holiday_name, 10, 1, days_delta=+1
+        )
+
+    def _add_eid_al_fitr_day_three(self, holiday_name) -> List[date]:
+        """
+        Add Eid al-Fitr Day Three.
+
+        https://en.wikipedia.org/wiki/Eid_al-Fitr
+        """
+        return self._add_islamic_calendar_holiday(
+            holiday_name, 10, 1, days_delta=+2
+        )
+
+    def _add_islamic_calendar_holiday(
+        self, holiday_name, month, day, days_delta=None
+    ) -> List[date]:
+        """
+        Add lunar calendar holiday.
+        """
+        dates = []
+
+        for dt in self._convert_islamic_to_gre(month, day):
+            if days_delta:
+                dt += rd(days=days_delta)
+
+            self._add_holiday(holiday_name, dt)
+            dates.append(dt)
+
+        return dates
+
+    def _add_islamic_new_year_day(self, holiday_name) -> List[date]:
+        """
+        Add Islamic New Year Day (last day of Dhu al-Hijjah).
+
+        The Islamic New Year, also called the Hijri New Year, is the day that
+        marks the beginning of a new lunar Hijri year, and is the day on which
+        the year count is incremented. The first day of the Islamic year is
+        observed by most Muslims on the first day of the month of Muharram.
+        https://en.wikipedia.org/wiki/Islamic_New_Year
+        """
+        return self._add_islamic_calendar_holiday(holiday_name, 1, 1)
+
+    def _add_isra_and_miraj_day(self, holiday_name):
+        """
+        Add Isra' and Mi'raj Day (27th day of 7th month).
+
+        https://en.wikipedia.org/wiki/Isra%27_and_Mi%27raj
+        """
+        return self._add_islamic_calendar_holiday(holiday_name, 7, 27)
+
+    def _add_mawlid_day(self, holiday_name) -> List[date]:
+        """
+        Add Mawlid Day (12th day of 3rd month).
+
+        Mawlid is the observance of the birthday of the Islamic prophet
+        Muhammad.
+        https://en.wikipedia.org/wiki/Mawlid
+        """
+        return self._add_islamic_calendar_holiday(holiday_name, 3, 12)
+
+    def _add_mawlid_day_two(self, holiday_name) -> List[date]:
+        """
+        Add Mawlid Day Two.
+
+        Mawlid is the observance of the birthday of the Islamic prophet
+        Muhammad.
+        https://en.wikipedia.org/wiki/Mawlid
+        """
+        return self._add_islamic_calendar_holiday(
+            holiday_name, 3, 12, days_delta=+1
+        )
+
+    def _convert_islamic_to_gre(self, month, day) -> List[date]:
+        """
+        Convert Islamic calendar date to Gregorian date(s).
+        """
+        return _islamic_to_gre(self._year, month, day)
```

Comment:
Need to somehow consider `self._year - 1` too, for multi-day holidays

Reply:
I'll look into it during per-country refactoring after the deep refactoring stuff is processed. I'm mostly focusing on not breaking existing logic here :)